androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/sideControls/VolumeControl.kt

```diff
@@ -0,0 +1,54 @@
+package com.profusion.androidenhancedvideoplayer.components.playerOverlay.sideControls
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import com.profusion.androidenhancedvideoplayer.components.playerOverlay.ControlsCustomization
+
+private const val MIN_VOLUME_VALUE = 0
+
+fun mapVolumeIconStateToIcon(
+    state: IconState,
+    customization: ControlsCustomization
+): @Composable () -> Unit {
+    return when (state) {
+        IconState.MED -> customization.volumeMedIconContent
+        IconState.OFF -> customization.volumeOffIconContent
+        IconState.HIGH -> customization.volumeHighIconContent
+    }
+}
+
+@Composable
+fun VolumeControl(
+    modifier: Modifier = Modifier,
+    volume: () -> Int,
+    maxVolumeValue: () -> Int,
+    setDeviceVolume: (Int) -> Unit,
+    customization: ControlsCustomization,
+    volumeMutableInteractionSource: MutableInteractionSource
+) {
+    val maxVolumeLocal = maxVolumeValue()
+    val halfVolume = maxVolumeLocal / 2
+    val lowRange = MIN_VOLUME_VALUE + 1..halfVolume
+    val highRange = halfVolume + 1..maxVolumeLocal
+    var volumeIconState by remember(volume()) {
+        mutableStateOf(
```

Comment:
This variable doesn't need to be a `State`, since it doesn't change. It can be just a `val` with `remember`